clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java

```diff
@@ -105,6 +105,23 @@ public class ConsumerConfig extends AbstractConfig {
     public static final String HEARTBEAT_INTERVAL_MS_CONFIG = CommonClientConfigs.HEARTBEAT_INTERVAL_MS_CONFIG;
     private static final String HEARTBEAT_INTERVAL_MS_DOC = CommonClientConfigs.HEARTBEAT_INTERVAL_MS_DOC;
 
+    /**
+     * <code>group.protocol</code>
+     */
+    public static final String GROUP_PROTOCOL_CONFIG = "group.protocol";
+    public static final String DEFAULT_GROUP_PROTOCOL = GroupProtocol.GENERIC.name().toLowerCase();
+    public static final String GROUP_PROTOCOL_DOC = "The rebalance protocol consumer should use.  We currently " +
+        "support \"generic\" or \"consumer\". If \"consumer\" is specified, then the consumer group protocol will be " +
+        "used.  Otherwise, the generic group protocol will be used.";
+
+    /**
+    * <code>group.remote.assignor</code>
+    */
+    public static final String REMOTE_ASSIGNOR_CONFIG = "group.remote.assignor";
+    public static final String DEFAULT_REMOTE_ASSIGNOR = null;
+    public static final String REMOTE_ASSIGNOR_DOC = "The server-side assignor to use. If no assignor is specified, " +
```

Comment:
Should we specify that the settings only applies when group.protocol = consumer?

Reply:
+1